compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -2367,7 +2367,7 @@ private void createVarRefAssignmentStmts(BLangTupleVarRef parentTupleVariable, B
      * This method creates a assignment statement and assigns and array expression based on the given indexExpr.
      *
      */
-    private void createSimpleVarRefAssignmentStmt(BLangVariableReference simpleVarRef, BLangBlockStmt parentBlockStmt,
+    private void createSimpleVarRefAssignmentStmt(BLangExpression simpleVarRef, BLangBlockStmt parentBlockStmt,
```

Comment:
Shall we rename this method and first param, as this function now support more than simpleVariableReferences

Reply:
Fixed in 278547735197ee003300179d272056e1b9251534